src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1360,17 +1360,10 @@ private unsafe void IntersectWithEnumerable(IEnumerable<T> other)
             int originalLastIndex = _lastIndex;
             int intArrayLength = BitHelper.ToIntArrayLength(originalLastIndex);
 
-            BitHelper bitHelper;
-            if (intArrayLength <= StackAllocThreshold)
-            {
-                int* bitArrayPtr = stackalloc int[intArrayLength];
-                bitHelper = new BitHelper(bitArrayPtr, intArrayLength);
-            }
-            else
-            {
-                int[] bitArray = new int[intArrayLength];
-                bitHelper = new BitHelper(bitArray, intArrayLength);
-            }
+            Span<int> span = stackalloc int[StackAllocThreshold];
```

Comment:
it's weird that we are duplicating the threshold all over again. Unrelated to this change of course.

Reply:
What do you mean?

Reply:
https://github.com/dotnet/corefx/blob/b7924b034f3a63cfb119180ddc88d61791e1c2f5/src/System.Collections/src/System/Collections/Generic/HashSet.cs#L58.

We have these constant thresholds all over the places where we use stackalloc. I know I added some in Regex for example. Shouldn't we have a common value?